contracts/NFTMarket.sol

```diff
@@ -507,6 +531,28 @@ contract NFTMarket is
         );
     }
 
+    function buyBackIfPriceIsLow(IERC721 _tokenAddress, uint256 _id, uint256 _price) private returns (bool) {
+        if(address(_tokenAddress) == address(characters)) {
+            uint8 characterLevel = characters.getLevel(_id);
+            int128 buyBackUSDValue = characterBuyBackNew + characterBuyBackPerLevel.mul(ABDKMath64x64.fromUInt(characterLevel));
+            uint256 buyBackTokenValue = usdToSkill(buyBackUSDValue);
+            if(buyBackTokenValue >= _price) {
+                _tokenAddress.safeTransferFrom(msg.sender, address(0), _id);
```

Comment:
The `.safeTransferFrom` function call won't allow passing the zero address as the destination. This will instead require the use of the `_burn` internal function in the NFT contract.